### PR TITLE
RHIROS-1158 implement filter by group for systems API

### PR DESCRIPTION
**Pull request by PreetiW**
##  filter by group_name  :boom:

implement filter by group_name for systems API
https://issues.redhat.com/browse/RHIROS-1158

Feel free to remove this section from PR description once done.

## Why do we need this change? :thought_balloon:

Please include the __context of this change__ here.

## Documentation update? :memo:

- [ ] Yes
- [x] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

### CODE REVIEW

File: ros/api/v1/hosts.py [hunk 1]
```diff
@@ -204,6 +204,9 @@ def build_system_filters():
                 else:
                     abort(400, message='Not a valid RHEL version')
             filters.append(System.operating_system.in_(modified_operating_systems))
+        if group_names := request.args.getlist('group_name'):
+            query = System.groups[0]['name'].astext.in_(group_names)
+            filters.append(query)
```
Comment (r14chandra) [inline ros/api/v1/hosts.py:209]:
```suggestion
            filters.append(System.groups[0]['name'].astext.in_(groups))
```

**Review by r14chandra — CHANGES_REQUESTED** [review]
Added minor nitpicks. Have a look.

File: ros/api/v1/hosts.py (hunk 1 above)
Comment (upadhyeammit) [reply ros/api/v1/hosts.py:209]: This won't check for all groups,
~~~
(Pdb) query = db.session.query(System.groups[0])
(Pdb) result = db.session.execute(query)
(Pdb) result.all()
[({'id': '7cbdaad6-d97e-4ed0-9095-ef07d73b4839', 'name': 'ros-test-1'},)]
(Pdb) query = db.session.query(System.groups[1])
(Pdb) result = db.session.execute(query)
(Pdb) result.all()
[({'id': '99999999-d97e-4ed0-9095-ef07d73b4839', 'name': 'ros-test-2'},)]
~~~

File: ros/api/v1/hosts.py [hunk 2]
```diff
@@ -204,6 +204,9 @@ def build_system_filters():
                 else:
                     abort(400, message='Not a valid RHEL version')
             filters.append(System.operating_system.in_(modified_operating_systems))
+        if group_names := request.args.getlist('group_name'):
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply ros/api/v1/hosts.py:207]: Hello @r14chandra, for the field name I referred to [inventory group documentation](https://docs.google.com/document/d/1O87MHvKlPEoH_DHwZw4htvwYlkGMiS-LwLQ_qKsZ5J0/edit#heading=h.1z0ggwl751hw) and implementation and based on that passing and accepting `group_name` in the query param made more sense to me. Have a look at the links and their host API and let me know your thoughts 🙌🏻

File: ros/api/v1/hosts.py (hunk 1 above)
Comment (PreetiW) [reply ros/api/v1/hosts.py]: @r14chandra for code consistency I updated the code by removing the query variable and pushed the changes in the changes in the latest commit. Have a look and let me know! 🙌🏻

Comment (PreetiW) [reply ros/api/v1/hosts.py]: @upadhyeammit Sorry I didn't get you, do you mean the query won't return all the systems with that `group_name`?

Comment (upadhyeammit) [reply ros/api/v1/hosts.py]: > @upadhyeammit Sorry I didn't get you, do you mean the query won't return all the systems with that `group_name`?

Actually one system can be in multiple groups and when you execute groups[0] it only checks with first element of array. The check should be for array of groups and not for just first element. Unfortunately I dont have answer for you on this one, as I am still playing with the queries.

File: ros/api/v1/hosts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (r14chandra) [reply ros/api/v1/hosts.py:207]: If that is the case then it is fine 👍🏻

**Review by r14chandra — APPROVED** [review]
Looks good!

File: ros/api/v1/hosts.py (hunk 1 above)
Comment (r14chandra) [reply ros/api/v1/hosts.py]: > Actually one system can be in multiple groups

Is it mentioned somewhere?

Comment (PreetiW) [reply ros/api/v1/hosts.py]: @upadhyeammit yes thats correct but for the MVP (Which we are focusing at the moment) the system can be part of only one group for now and hence I wrote the query like this. I feel let's move ahead if it looks good for now and when inventory will start supporting multiple groups feature we will also adapt to that?
[Here is the document link](https://docs.google.com/document/d/1_zWWRpunieVbvuR6SMjP2fReA1YOS1XruLQaTnYjb6A/edit#heading=h.rjwpur2zwfgz) where I read about the single group scope, have a look and let me know your thoughts 🙌🏻

Comment (upadhyeammit) [reply ros/api/v1/hosts.py]: Wow! thats good to know, i have been battling to accommodate the multiple groups! Yeah then your changes makes sense!

**Review by upadhyeammit — APPROVED** [review]

File: ros/api/v1/hosts.py (hunk 1 above)
Comment (PreetiW) [reply ros/api/v1/hosts.py]: great, thanks for clarifying it out @upadhyeammit and let me know if any more questions :)

**Pull request merged**